prov/verbs/src/ep_rdm/verbs_tagged_ep_rdm_states.c

```diff
@@ -1433,9 +1442,9 @@ fi_ibv_rdm_rma_post_ready(struct fi_ibv_rdm_request *request, void *data)
 	request->rest_len -= seg_cursize;
 	request->post_counter++;
 	FI_IBV_RDM_INC_SIG_POST_COUNTERS(request->minfo.conn, p->ep_rdm);
-	int ret = ibv_post_send(request->minfo.conn->qp[0], &wr, &bad_wr);
 
-	if (request->rest_len) {
+	ret = ibv_post_send(request->minfo.conn->qp[0], &wr, &bad_wr);
```

Comment:
This fixes possible bug if ibv_post_send fails but the branch `(request->rest_len && !ret)` takes true, under this if-test, the `ret` variable is overwritten (i.e. the `ret` variable can be set, but never referenced with previous value)